src/settings/menuoptions.cpp

```diff
@@ -0,0 +1,93 @@
+#include "menuoptions.h"
+#include <QDebug>
+
+#include <QVBoxLayout>
+MenuOptions::MenuOptions(QWidget *pParent) : QDialog (pParent)
+{
+    setModal(true);
+    int width = 850;
+    int height = 280;
+    setMinimumWidth(width);
+    setMaximumWidth(width);
+    setMinimumHeight(height);
+    setMaximumHeight(height);
+    setWindowTitle("Settings");
+
+    tabWidget = new QTabWidget;
+
+    tabEnvironment = new TabEnvironment(this);
+    tabTextEditor = new TabTextEditor(this);
+    tabCpp = new TabCpp(this);
+    tabRepository = new TabRepository(this);
+    tabConnection = new TabConnection(this);
+
+
+    tabWidget->addTab(tabEnvironment, "Environment");
+    tabWidget->addTab(tabTextEditor, "Text Editor");
+    tabWidget->addTab(tabCpp, "C++");
+    tabWidget->addTab(tabRepository, "Repository");
+    tabWidget->addTab(tabConnection, "Connection");
+
+    buttonBox = new QDialogButtonBox(QDialogButtonBox::Ok
+           | QDialogButtonBox::Apply | QDialogButtonBox::Cancel);
+
+    connect(buttonBox, &QDialogButtonBox::clicked, this, &MenuOptions::onBtnClicked);
+
+    QVBoxLayout *mainLayout = new QVBoxLayout;
+    mainLayout->addWidget(tabWidget);
+    mainLayout->addWidget(buttonBox);
+    setLayout(mainLayout);
+    show();
+}
+
+void MenuOptions::onBtnApplyClicked()
+{
+    //      APPLY ALL CHANGES IN TABS
+    // tab text editor
+    if (tabTextEditor->newValuesToChange.size())
+    {
+        emit valuesForEditor(tabTextEditor->newValuesToChange);
+
+        std::map<QString, QString>::iterator it;
+        QSettings s;
+        // check if font size was changed
+        it = tabTextEditor->newValuesToChange.find("editorFontSize");
+        if (it != tabTextEditor->newValuesToChange.end())
```

Comment:
1)you do this check every time, but I don't think it's not necessary. You're sure that fields "editorFontSize", "editorStyle" and "editorFont" exists, then std::map.find will never return the map.end. If you wanna do the check are the date read from JSON correctly or not, don't do it here. 
2)you also know the order of placement of the elements in your map. You have only 3 elements, you don't need to find them every time